src/toBeSorted/d_emitter.cpp

```diff
@@ -0,0 +1,1298 @@
+        g = mgr.getfield_0x2F18() * g1 * scaleG;
+        b = mgr.getfield_0x2F1C() * b1 * scaleB;
+    } else {
+        r = r1 * scaleR;
+        g = g1 * scaleG;
+        b = b1 * scaleB;
+    }
+
+    emitter->setGlobalPrmColor(r, g, b);
+
+    if (color2 != nullptr) {
+        r2 = color2->r;
+        g2 = color2->g;
+        b2 = color2->b;
+    }
+
+    if (plltIdx1 > 0) {
+        if ((flags & 0x1000) != 0) {
+            const PaletteEAF_smol_entry &entry = mgr.getSmallEAF(plltIdx1 - 1, plltIdx2);
+            r2 = entry.field_0x04.r;
+            g2 = entry.field_0x04.g;
+            b2 = entry.field_0x04.b;
+        } else if ((flags & 0x2000) != 0) {
+            const PaletteEAF_smol_entry &entry = mgr.getSmallEAF(plltIdx1 - 1, plltIdx2);
+            r2 = entry.field_0x0C.r;
+            g2 = entry.field_0x0C.g;
+            b2 = entry.field_0x0C.b;
+        }
+    }
+
+    if (factor != 0) {
+        r = mgr.getfield_0x2F14() * r2 * scaleR;
+        g = mgr.getfield_0x2F18() * g2 * scaleG;
+        b = mgr.getfield_0x2F1C() * b2 * scaleB;
+    } else {
+        r = r2 * scaleR;
+        g = g2 * scaleG;
+        b = b2 * scaleB;
+    }
+
+    emitter->setGlobalEnvColor(r, g, b);
+}
+
+void EffectsStruct::removeFromActiveEmittersList() {
+    if (dJEffManager_c::sPlayingEffectsList.GetPosition(this) != dJEffManager_c::sPlayingEffectsList.GetEndIter()) {
+        dJEffManager_c::sPlayingEffectsList.remove(this);
+    }
+}
+
+bool EffectsStruct::createEffect(
+    bool bFlags, u16 resourceId, const mVec3_c &pos, const mAng3_c *rot, const mVec3_c *scale, const GXColor *c1,
+    const GXColor *c2
+) {
+    if (!bFlags && canReuse(resourceId)) {
+        setPosRotScale(pos, rot, scale);
+        s32 idx1 = 0;
+        s32 idx2 = 0;
+        getOwnerPolyAttrs(&idx1, &idx2);
+        loadColors(c1, c2, idx1, idx2);
+        onFlag(EMITTER_0x1);
+    } else {
+        remove(false);
+        s32 idx1 = 0;
+        s32 idx2 = 0;
+        getOwnerPolyAttrs(&idx1, &idx2);
+        if (createEmitters(resourceId, pos, rot, scale, c1, c2, idx1, idx2)) {
+            addToActiveEmittersList(resourceId, bFlags);
+        }
+    }
+
+    return hasEmitters();
+}
+
+bool EffectsStruct::createEffect(
+    bool bFlags, u16 resourceId, const mMtx_c &transform, const GXColor *c1, const GXColor *c2
+) {
+    if (!bFlags && canReuse(resourceId)) {
+        s32 idx1 = 0;
+        s32 idx2 = 0;
+        getOwnerPolyAttrs(&idx1, &idx2);
+        loadColors(c1, c2, idx1, idx2);
+        onFlag(EMITTER_0x1);
+    } else {
+        remove(false);
+        s32 idx1 = 0;
+        s32 idx2 = 0;
+        getOwnerPolyAttrs(&idx1, &idx2);
+        if (createEmitters(resourceId, mVec3_c::Zero, nullptr, nullptr, c1, c2, idx1, idx2)) {
+            addToActiveEmittersList(resourceId, bFlags);
+        }
+    }
+    setTransform(transform);
+    return hasEmitters();
+}
+
+s32 dJEffManager_c::polyAttrsToGroundEffectIdx(s32 polyAttr0, s32 polyAttr1) {
```

Comment:
I am assuming this has to do with the `dBgPc_Attribute` form `include\d\col\bg\d_bg_pc.h` ?
```suggestion
// TODO: Document PolyAttrs
s32 dJEffManager_c::polyAttrsToGroundEffectIdx(s32 polyAttr0, s32 polyAttr1) {
```